### ISSUE TITLE
New components from 3.0 preventing useVersion from compiling code in 3.0.4.

### ISSUE THREAD

**Issue by dvbridges** [psychopy/psychopy#2286]
In 3.0.4 (Py3: WIndows 10), using ```useVersion``` to compile code from versions < 3.0 throws the following error for:

* joystick
* pump
* slider
* variable

```
if module.__file__.endswith('.pyc'):
    AttributeError: module 'psychopy.experiment.components.variable' has no attribute '__file__'
```

The error occurs because the component folders in the ```versions``` directory, that do not exist in previous versions, contain only \_pycache_ folders (and no \_\_init\_\_.py files). So, perhaps this is an issue deleting folder-containing component folders when useVersion is called.

**Comment by peircej** [comment]
I wonder if the problem was just that I hadn't pushed 3.0.4 up to the versions repo. That's now done. Could you try again?

**Comment by dvbridges** [comment]
Great, all fixed. Just deleted the old versions folder just to make sure, reopened PsychoPy and compiled the experiment using 1.90.0 and all is well.

**Comment by dvbridges** [comment]
Closing because fixed - Jon just needed to push 3.0.4 up to the versions repo.
